### PR TITLE
Added Support for Nim LSP and Grammar.

### PR DESCRIPTION
**Pull request by V0idMatr1x**

### CODE REVIEW

File: languages.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ indent = { tab-width = 4, unit = "    " }
 [language.auto-pairs]
 '(' = ')'
 '{' = '}'
-'[' = ']'
+'[' = ']'
```
Comment (the-mikedavis) [inline languages.toml:18]: can you revert this change?

File: runtime/runtime [hunk 2]
```diff
@@ -0,0 +1 @@
+/Users/v0id/helix/runtime
```
Comment (the-mikedavis) [inline runtime/runtime:1]: and this one

File: languages.toml [hunk 3]
```diff
@@ -1409,3 +1409,16 @@ injection-regex = "verilog"
 [[grammar]]
 name = "verilog"
 source = { git = "https://github.com/andreytkachenko/tree-sitter-verilog", rev = "514d8d70593d29ef3ef667fa6b0e504ae7c977e3" }
+
+[[language]]
+name = "nim"
+scope = "source.nim"
+file-types = ["nim", "nimble"]
```
Comment (the-mikedavis) [inline languages.toml:1416]:
```suggestion
file-types = ["nim", "nims", "nimble"]
```

to add support for nimscript

**Review by the-mikedavis — CHANGES_REQUESTED** [review]

File: runtime/runtime (hunk 2 above)
Comment (V0idMatr1x) [reply runtime/runtime:1]: How do I remove this?

**Comment by archseer** [comment]
The grammar itself isn't useful without the queries so this PR should wait until those are added.

**Comment by V0idMatr1x** [comment]
> The grammar itself isn't useful without the queries so this PR should wait until those are added.

Understandable, I am working on the Highlight queries at the moment trying to figure that part out.

**Comment by V0idMatr1x** [comment]
Highlight queries seem to be functional now; is it okay to merge?

Screen Shot 2022-05-31 at 8 06 18 PM
